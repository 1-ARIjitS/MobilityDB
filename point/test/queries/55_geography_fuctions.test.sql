-------------------------------------------------------------------------------

-- Function added to PostGIS to interpolate a point along a geography line
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.0, false, true));
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 1.0, false, true));
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.1, false, false));
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.1, false, true));
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.1, true, true));
-- EMPTY
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring empty', 0.1, false, true));
/* Errors */
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 2, false, true));
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Point(4.35 50.85)', 0.5, false, true));

-------------------------------------------------------------------------------