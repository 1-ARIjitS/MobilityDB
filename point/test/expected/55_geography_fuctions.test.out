SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.0, false, true));
     st_astext     
-------------------
 POINT(4.35 50.85)
(1 row)

SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 1.0, false, true));
         st_astext          
----------------------------
 POINT(37.617222 55.755833)
(1 row)

SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.1, false, false));
                st_astext                 
------------------------------------------
 POINT(7.21618346298142 51.7185560475017)
(1 row)

SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.1, false, true));
                                                                                                                                                                            st_astext                                                                                                                                                                            
-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 MULTIPOINT(7.21618346298142 51.7185560475017,10.2202139911455 52.5223388215793,13.3546063535787 53.2525167375086,16.6081132068154 53.9008087360718,19.9655715034098 54.4598130904273,23.4080093014734 54.9233351500471,26.9130586508111 55.2866859815338,30.4556818372668 55.5469218873619,34.0091691619729 55.7029975877171,37.5463175956809 55.7558138476789)
(1 row)

SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 0.1, true, true));
                                                                                                                                                                    st_astext                                                                                                                                                                    
-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
 MULTIPOINT(7.2220985969596 51.7202434531659,10.2325875123559 52.5254342648845,13.3739250151343 53.2566887167436,16.6347759217365 53.9056793854838,19.9998576749004 54.4649673133545,23.4500503284172 54.9283329391853,26.9628165216683 55.2910773280783,30.512938399544 55.5502632631239,34.0735284866234 55.7048686660918,37.617222 55.755833)
(1 row)

SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring empty', 0.1, false, true));
  st_astext  
-------------
 POINT EMPTY
(1 row)

/* Errors */
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Linestring(4.35 50.85, 37.617222 55.755833)', 2, false, true));
ERROR:  line_interpolate_point: 2nd arg isn't within [0,1]
SELECT ST_AsText(st_lineinterpolatepoints(geography 'Point(4.35 50.85)', 0.5, false, true));
ERROR:  line_interpolate_point: 1st arg isn't a line
